src/networking/nc/mod.rs

```diff
@@ -0,0 +1,958 @@
+            AF_UNSPEC
+        };
+
+        if positionals.len() == 1 {
+            if family == AF_UNIX {
+                host = String::from(positionals[0]);
+                uport = String::new();
+            } else {
+                if !lflag {
+                    return mesaerr_result(msg);
+                }
+                uport = String::from(positionals[0]);
+            }
+        } else if positionals.len() >= 2 {
+            host = String::from(positionals[0]);
+            uport = String::from(positionals[1]);
+        } else {
+            return mesaerr_result(msg);
+        }
+
+        if lflag && s_addr.is_some() {
+            return mesaerr_result("cannot use -s and -l");
+        }
+        if lflag && pflag {
+            return mesaerr_result("cannot use -p and -l");
+        }
+        if lflag && zflag {
+            return mesaerr_result("cannot use -z and -l");
+        }
+        if !lflag && kflag {
+            return mesaerr_result("must use -l with -k");
+        }
+
+        if !uport.is_empty() {
+            portlist = build_ports(&uport)?;
+        }
+
+        if matches.is_present("i") {
+            let sec = matches.value_of("i").unwrap().parse::<u64>().unwrap();
+            interval = Some(Duration::new(sec, 0));
+        }
+
+        if matches.is_present("w") {
+            let sec = matches.value_of("w").unwrap().parse::<u64>().unwrap();
+            timeout = Some(Duration::new(sec, 0));
+        }
+
+        let mut unix_dg_tmp_socket = String::new();
+
+        /* Get name of temporary socket for unix datagram client */
+        if family == AF_UNIX && uflag && !lflag {
+            unix_dg_tmp_socket = if s_addr.is_some() {
+                s_addr.clone().unwrap()
+            } else {
+                let nf = NamedTempFile::new()?;
+                let path = String::from(nf.path().to_str().unwrap());
+                path
+            };
+        }
+
+        let ret = NcOptions {
+            dflag: matches.is_present("d"),
+            iflag: matches.is_present("i"),
+            interval: interval,
+            lflag: matches.is_present("l"),
+            unixflag: matches.is_present("U"),
+            uflag: uflag,
+            host: host,
+            portlist: portlist,
+            family: family,
+            kflag: matches.is_present("k"),
+            s_addr: s_addr,
+            pflag: pflag,
+            vflag: matches.is_present("v"),
+            timeout: timeout,
+            unix_dg_tmp_socket: unix_dg_tmp_socket,
+            zflag: zflag,
+        };
+
+        return Ok(ret);
+    }
+}
+
+fn remove_item<T: Eq+Debug>(v: &mut Vec<T>, item: T) {
+    debug_info(&format!("remove_item {:?}", item));
+    match v.iter().position(|t| *t == item) {
+        Some(i) => v.remove(i),
+        None => return
+    };
+}
+
+struct NcCore<'a> {
+    sock: &'a mut Socket,
+    opts: &'a NcOptions,
+    poll: Poll,
+    net_interest: Ready,
+    event_stdin: EventedFd<'a>,
```

Comment:
Because of `UtilSetup`, this will need to check `setup.input().raw_fd()` rather than just using `stdin`.  If the result is `None`, we can just assume there is no input (it's either something like `Vec<u8>`, which has no file descriptor, or `io::Empty`).

Reply:
done